app/controllers/admin/site_texts_controller.rb

```diff
@@ -41,8 +45,15 @@ def index
       end
     end
 
-    extras[:has_more] = true if results.size > 50
-    render_serialized(results[0..49], SiteTextSerializer, root: 'site_texts', rest_serializer: true, extras: extras, overridden_keys: overridden_keys)
+    per_page = 50
+    page = params[:page].to_i
+    first = page * per_page
+    last = first + per_page
+
+    raise Discourse::InvalidParameters.new(:page) if page < 0
```

Comment:
If `page < 0`, then you have calculated `first` and `last` for nothing 😉

Reply:
I wanted to keep all computations in one place hoping to make things more clear. 😊